pkg/manifests/manifests.go

```diff
@@ -2385,16 +2459,38 @@ func (f *Factory) PrometheusOperatorUserWorkloadDeployment() (*appsv1.Deployment
 	return d, nil
 }
 
-func (f *Factory) setTLSSecurityConfiguration(args []string, tlsCipherSuitesArg string, minTLSversionArg string) []string {
-	cipherSuites := strings.Join(crypto.OpenSSLToIANACipherSuites(f.APIServerConfig.GetTLSCiphers()), ",")
+type minTLSVersioner interface {
+	MinTLSVersion() string
+}
+
+type minTLVersionEnforcer struct {
+	atLeast uint16
+	inner   minTLSVersioner
+}
+
+func (m *minTLVersionEnforcer) MinTLSVersion() string {
+	v := m.inner.MinTLSVersion()
+	if crypto.TLSVersionOrDie(v) < m.atLeast {
+		return crypto.TLSVersionToNameOrDie(m.atLeast)
+	}
+
+	return v
+}
+
+func (f *Factory) setTLSSecurityConfigurationWithMinTLSVersion(args []string, tlsCipherSuitesArg string, minTLSversionArg string, versioner minTLSVersioner) []string {
+	cipherSuites := strings.Join(crypto.OpenSSLToIANACipherSuites(f.APIServerConfig.TLSCiphers()), ",")
 	args = setArg(args, tlsCipherSuitesArg, cipherSuites)
 
-	minTLSVersion := f.APIServerConfig.GetMinTLSVersion()
+	minTLSVersion := versioner.MinTLSVersion()
 	args = setArg(args, minTLSversionArg, string(minTLSVersion))
```

Comment:
we don't need to cast to `string` do we?

Reply:
correct